File: tutorial-for-assignment1/java/Lookfor.java
import java.io.*;

public class Lookfor {
    public static void main(String[] args) {
        if(args.length != 2) {
            System.out.println("Usage: <filename> <keyword>");
            System.exit(0);
        }

        String filename = args[0];
        String keyword  = args[1];

        InputStream in;
        try {
            in = new FileInputStream(filename);
            BufferedReader reader = new BufferedReader(new InputStreamReader(in));
            String line = reader.readLine();
            long start = System.currentTimeMillis();
            int count = 0;
            while(line != null) {
                if(line.contains(keyword)) {
                    System.out.println(line);
                    count += 1;
                }
                line = reader.readLine();
            }
            float dur = (System.currentTimeMillis() - start) / 1000f;
            System.out.printf("Found %d instances in %.4f seconds\n",
                count,
                dur);
            reader.close();
        } catch(Exception e) {
            System.out.printf("Cannot open file \"%s\"\n", filename);
            System.exit(0);
        }

    }
}

File: tutorial-for-assignment1/go/lookfor.go
package main

import (
    "fmt"
    "os"
    "bufio"
    "strings"
    "time"
)

func main() {
    if len(os.Args) != 3 {
        fmt.Println("Usage <filename> <keyword>")
        os.Exit(0)
    }
    filename, keyword := os.Args[1], os.Args[2]

    f, err := os.Open(filename)

    if err != nil {
        fmt.Printf("Cannot open file \"%s\"\n", filename)
        os.Exit(0)
    }

    reader := bufio.NewReader(f)
    start := time.Now()
    count := 0
    for line, err := reader.ReadString('\n');
        err == nil;
        line, err = reader.ReadString('\n') {
        if strings.Contains(line, keyword) {
            fmt.Print(line)
            count += 1
        }
    }
    duration := time.Now().Sub(start)
    fmt.Printf("Found %d instances in %.4f seconds\n",
        count,
        float32(duration.Nanoseconds()) / 10E9)
}
